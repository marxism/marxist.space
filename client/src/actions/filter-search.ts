export const SET_TAG = 'SET_TAG'
export const CLEAR_TAGS = 'CLEAR_TAGS'

const setTagAction = (tag: string) => ({
  type: SET_TAG,
  payload: tag,
})

const clearTagsAction = () => ({
  type: CLEAR_TAGS,
})

export const setTag = (tag: string) => (dispatch) => {
  dispatch(setTagAction(tag))
}

export const clearTags = () => (dispatch) => {
  dispatch(clearTagsAction())
}

export const SET_SEARCH = 'SET_SEARCH'
export const CLEAR_SEARCH = 'CLEAR_SEARCH'

const setSearchAction = (search: string) => ({
  type: SET_SEARCH,
  payload: search,
})

const clearSearchAction = () => ({
  type: CLEAR_SEARCH,
})

export const setSearch = (search: string) => (dispatch) => {
  dispatch(setSearchAction(search))
}

export const clearSearch = () => (dispatch) => {
  dispatch(clearSearchAction())
}
