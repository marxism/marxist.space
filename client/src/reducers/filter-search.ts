/* eslint-disable unicorn/no-keyword-prefix */

import {
  SET_TAG,
  CLEAR_TAGS,
  SET_SEARCH,
  CLEAR_SEARCH,
} from '../actions/filter-search'

const initialState = { search: '', tags: [] }

export default (state = initialState, action) => {
  switch (action.type) {
    case SET_TAG:
      const newTag = action.payload
      // @ts-ignore
      const newTags = state.tags.includes(newTag)
        ? state.tags.filter((t) => t !== newTag)
        : [...state.tags, newTag]

      return { ...state, tags: newTags, search: '' }
    case CLEAR_TAGS:
      return { ...state, tags: [] }
    case SET_SEARCH:
      return { ...state, search: action.payload, tags: [] }
    case CLEAR_SEARCH:
      return { ...state, search: '' }
    default:
      return state
  }
}
