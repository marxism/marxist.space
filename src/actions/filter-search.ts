export const SET_TAG = 'SET_TAG'
export const CLEAR_TAGS = 'CLEAR_TAGS'

export const setTagAction = (tag: string) => ({
  type: SET_TAG,
  payload: tag,
})

export const clearTagsAction = () => ({
  type: CLEAR_TAGS,
})

export const setTag = (tag: string) => (dispatch) => {
  dispatch(setTagAction(tag))
}

export const clearTags = () => (dispatch) => {
  dispatch(clearTagsAction())
}

export const SET_SEARCH = 'SET_SEARCH'
export const CLEAR_SEARCH = 'CLEAR_SEARCH'

export const setSearchAction = (search: string) => ({
  type: SET_SEARCH,
  payload: search,
})

export const clearSearchAction = () => ({
  type: CLEAR_SEARCH,
})

export const setSearch = (search: string) => (dispatch) => {
  dispatch(setSearchAction(search))
}

export const clearSearch = () => (dispatch) => {
  dispatch(clearSearchAction())
}
