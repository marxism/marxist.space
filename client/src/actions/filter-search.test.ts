/* eslint-env jest */

import * as filterSearchActions from './filter-search'

describe('filterSearch actions', () => {
  test('setTagAction', () => {
    expect(filterSearchActions.setTagAction('one')).toMatchSnapshot()
  })

  test('clearTagsAction', () => {
    expect(filterSearchActions.clearTagsAction()).toMatchSnapshot()
  })

  test('setSearchAction', () => {
    expect(filterSearchActions.setSearchAction('one')).toMatchSnapshot()
  })

  test('clearSearchAction', () => {
    expect(filterSearchActions.clearSearchAction()).toMatchSnapshot()
  })
})
