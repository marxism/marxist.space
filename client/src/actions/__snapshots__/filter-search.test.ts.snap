// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filterSearch actions clearSearchAction 1`] = `
Object {
  "type": "CLEAR_SEARCH",
}
`;

exports[`filterSearch actions clearTagsAction 1`] = `
Object {
  "type": "CLEAR_TAGS",
}
`;

exports[`filterSearch actions setSearchAction 1`] = `
Object {
  "payload": "one",
  "type": "SET_SEARCH",
}
`;

exports[`filterSearch actions setTagAction 1`] = `
Object {
  "payload": "one",
  "type": "SET_TAG",
}
`;
